nemo_curator/modules/task.py

```diff
@@ -318,7 +320,15 @@ def _remove_ngrams_partition(self, partition, task_ngrams, ngrams_freq_sorted):
 
         partition[self.text_field] = split_text
         filtered_partition = partition[valid_documents_mask]
-        return filtered_partition.explode(self.text_field, ignore_index=True)
+        exploded_partition = filtered_partition.explode(
+            self.text_field, ignore_index=True
+        )
+        # After exploding, the string datatype can become an "object" type
+        exploded_partition[self.text_field] = exploded_partition[
+            self.text_field
+        ].astype(text_type)
+
+        return exploded_partition
```

Comment:
Is this a pandas bug that we somehow exposed or maybe expected behavior?

Reply:
I can't find anything on the pandas side of things that would suggest it's a bug. I doubt it is, but my understanding could be incorrect. In my head, the resulting column from an explode operation should probably be an `object` type since lists can contain anything. Though, not sure if pandas says somewhere that it should automatically convert the column datatype if all elements of the list are all strings/ints or whatever.

Reply:
It's only in our case that we can guarantee all the list elements will be strings. And since we make the promise to Dask in the `from_delayed` call on line 278 that the resulting dataframe will still have the text field as a string, we need to do this conversion.